### PR TITLE
wip: not filter the only EmptyStatement

### LINKED ISSUE #347: Minifies single `EmptyStatement` is not idempotent
Our [tester](https://github.com/Boshen/oxc/blob/48e524ddcfb9570139cd74b9348707f1c4c5656f/tasks/coverage/src/minifier.rs#L79) require minifying twice idempotent, but when minify `if(true);`:
1. `if(true);` => `if(true){;}`
2. `if(true){;}` => `if(true){}`

and it causes [this test case](https://github.com/tc39/test262/blob/4fc741b008abdb6adbc89ba4d5a250ecb69edd8a/test/annexB/language/function-code/if-stmt-else-decl-func-block-scoping.js#L40) failed:

```js
var initialBV, currentBV, varBinding;

(function() {
  

  if (false) ; else function f() { initialBV = f; f = 123; currentBV = f; return 'decl'; }

  varBinding = f;
  f();
}());
```

### PR DESCRIPTION
**Pull request by mysteryven**
closes #347 

I found this issue when I solving [problem with parentheses inside minifier](https://github.com/Boshen/oxc/issues/338), It's not about parentheses, so I pull a single Request for it. I hope it is the right way to split it.

With Boshen's suggestion, I minifiy statements like `if(true){;}` to `if(true);`, But I don't think my implement is good. Is it can have another better solution?

### CODE REVIEW

File: crates/oxc_minifier/src/compressor/mod.rs [hunk 1]
```diff
@@ -182,7 +182,9 @@ impl<'a> Compressor<'a> {
 
 impl<'a, 'b> VisitMut<'a, 'b> for Compressor<'a> {
     fn visit_statements(&mut self, stmts: &'b mut Vec<'a, Statement<'a>>) {
-        stmts.retain(|stmt| !self.drop_empty(stmt) && !self.drop_debugger(stmt));
+        if stmts.len() > 1 {
```
Comment (Boshen) [inline crates/oxc_minifier/src/compressor/mod.rs:185]: I don't think this is correct, we can safely drop all statements and leave it as a empty vec.

File: crates/oxc_minifier/src/printer/mod.rs [hunk 2]
```diff
@@ -233,8 +233,13 @@ impl Printer {
         }
     }
 
-    pub fn print_block1(&mut self, stmt: &BlockStatement<'_>) {
-        self.print(b'{');
+    pub fn print_block1(&mut self, stmt: &BlockStatement<'_>, can_omit_braces: bool) {
```
[the remark this answers is not in the document]
Comment (Boshen) [reply crates/oxc_minifier/src/printer/mod.rs:236]: The printer shouldn't have any minifying logic I think, let me think about the best way to do this.

**Comment by Boshen** [comment]
https://github.com/Boshen/oxc/blob/48e524ddcfb9570139cd74b9348707f1c4c5656f/crates/oxc_minifier/src/printer/gen.rs#L108-L119

This looks pretty wrong to me, just change it to print the block instead of the weird if-else statement and see what happens?

**Comment by mysteryven** [comment]
[**@Boshen**](https://github.com/Boshen) commented on [May 10, 2023, 11:24 PM GMT+8](https://github.com/Boshen/oxc/pull/348#issuecomment-1542403618 "2023-05-10T15:24:16Z - Replied by Github Reply Comments"):
> Is this the bug?
> 
> `if (true) ; else {}` -> `if(true){;} else {}`
> 
> _Originally posted by @Boshen in https://github.com/Boshen/oxc/issues/348#issuecomment-1542403618_

I apologize for not explaining the issue clear. Our tester assumes minification is an idempotent. However, if  we minify the code`if(true); else {}` called s1, the Printer will output source_text1: 

```js
if (!0) {
    ;
}  else {
}
```
If we minify source_text1 next, it will drain the `EmptyStatement`, then will output source_text2.

```js
if (!0) {
}  else {
}
```

 source_text1 not equal with source_text2, so our tester failed.


> I don't think this is correct, we can safely drop all statements and leave it as a empty vec.

When we drop all statements, I have no idea how to pass our tester pass when we encounter test cases like above. Or maybe we can convert `if(true);` to `if(true){}`👀

**Pull request closed without merging**